base/challenge-skeleton/README.md

```diff
@@ -1,52 +1,61 @@
+* `pow.conf`: Contains the difficulty of the proof-of-work, 0 means disabled.
+* `advanced`: The Kubernetes config used to deploy the challenge. E.g. you can add a tmpfs here or change the port.
 
-The config directory holds a few configuration files:
+### /challenge
 
-* chal.conf: Configure if the challenge should be deployed to the cluster, if it's publicly accessible and if it has a healthcheck.
-* pow.conf: Contains the difficulty of the proof-of-work, 0 means disabled.
-* advanced: The kubernetes config used to deploy the challenge. E.g. you can add a tmpfs here or change the port.
+The `challenge` directory contains the challenge binary and anything else related to the challenge:
 
-## /challenge
+* `image/`: Dockerfile and files of the challenge image.
+* `Makefile`: Edit this file if you need to run code before building the Docker image, e.g. for building the challenge from source.
 
-The challenge binary and anything else challenge related goes in here.
+### /healthcheck
 
-* image/: Dockerfile and files of the challenge image.
-* Makefile: You can edit this file if you need to run code before building the docker image, e.g. for building the challenge from source.
+The `healthcheck` directory is optional. If you don't want to write a healthcheck, feel free to delete the directory. However, we strongly recommend that you implement a healthcheck :).
 
-## /healthcheck
+* `image/`: Dockerfile and files of the healthcheck.
+* `Makefile`: Edit this file if you need to build any custom files before running your healthcheck.
 
-The healthcheck directory is optional. If you don't want to write a healthcheck, feel free to delete it. You really should have healthchecks though :).
 
-* image/: Dockerfile and files of the healthcheck.
-* Makefile: same as for the challenge directory.
+#### Healthcheck
+
+Important details regarding healthchecks:
+
+* Run the healthcheck on a webserver on port 45281 that responds to `/healthz` requests.
+ * The healthcheck returns 200 if the challenge is healthy, otherwise an error (e.g. 400).
+ * You can find an example webserver in `healthcheck/image/healthz.py`.
+* The base image comes preloaded with pwntools.
+* If your exploit is written in Python, drop it in `healthcheck/image/doit.py`.
 
-# API contract
+## API contract
 
-Here are the requirements what this directory must look like to work with kCTF:
+Ensure your setup fulfills the following requirements to ensure it works with kCTF:
 
-* You can do pretty much whatever you want in the challenge dir but:
-  * You need to have a Makefile with the .gen/docker-image target that builds a docker image.
-  * We strongly recommend to use nsjail in all challenges.
-* Your challenge will receive connections on port 1337.
+* In your cmdline, call `kctf_setup` as the first command.
+* You can do pretty much whatever you want in the `challenge` directory but:
+  * You need to have a `Makefile` with the `.gen/docker-image` target that builds a Docker image.
```

Comment:
Same as before, can we add an example? And explain what that file should contain
```Makefile
.gen/docker-image:
docker build . >$@
```

Reply:
wouldn't it be better to link to the Makefile instead?

Reply:
sg

Reply:
https://github.com/google/kctf/blob/master/base/challenge-skeleton/Makefile

Reply:
@goedi02 do you think it's clear? Not everyone might be familiar with Makefiles. If we linked you to that file would you know what we mean by this sentence?

Reply:
@sirdarckcat not a 100% clear to me. In the file we link to, I can see ".gen/docker-image" is already there. But I assume I need to customize/modify that for my use case? We ought to emphasize the action:

"In the [Makefile](https://github.com/google/kctf/blob/master/base/challenge-skeleton/challenge/Makefile): Ensure the `.gen/docker-image` target that builds a Docker image is configured according to your setup."

- can we assume people know what they need to change when configuring? Or should we add more detail on that?

Reply:
I think most users won't need to change it, so maybe we can mention that by default the makefile is configured to build a docker image?

Reply:
Ok, rephrased this to: "In the Makefile, the .gen/docker-image target that builds a Docker image must be configured. The configuration is set by default and usually does not require changes."